### PR TITLE
Fixing flaky tests for windows 7 32bit

### PR DESCRIPTION
**Pull request by narph**
Will update soon
## What does this PR do?

After investigation we see 2 main issues:

1. `constant .... overflows int`

Because the `int` value is overflowed, the fix is to specify the `int64` type instead.

2. Tests failing consistently most likely due to `expfmt` package. This happens in function https://github.com/elastic/beats/blob/master/metricbeat/helper/prometheus/prometheus.go#L72
When trying to read and parse the data, lines for example:

```
etcd_disk_wal_fsync_duration_seconds_bucket{le="0.002"} 0	
etcd_disk_wal_fsync_duration_seconds_bucket{le="0.004"} 0	
```
are not considered.

## Why is it important?

Should fix/skip the following:
(etcd failing tests were previously fixed)
image

image

Python tests

image

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Closes https://github.com/elastic/beats/issues/19835

### CODE REVIEW

**Comment by narph** [comment]
@v1v , most of tests are failing consistently most likely due to `expfmt` package. This happens in function https://github.com/elastic/beats/blob/master/metricbeat/helper/prometheus/prometheus.go#L72
 When trying to read and parse the data, lines for example:

etcd_disk_wal_fsync_duration_seconds_bucket{le="0.002"} 0	
etcd_disk_wal_fsync_duration_seconds_bucket{le="0.004"} 0	

are not considered.

If we change the value `0` to `1` then everything is fine. This PR fixes the `etcd` module tests. I have upgraded the package but error still consists. I am not sure there is a solution for this tests for 32 bit os's.
What should we do in this case? Update test files? Make sure we skip these specific tests for this os?

**Comment by narph** [comment]
running the tests again 

/test

File: metricbeat/module/ceph/cluster_status/cluster_status_test.go [hunk 1]
```diff
@@ -94,10 +94,13 @@ func TestFetchEventContents(t *testing.T) {
...
+	var avb int64 = 9965821952
+	var tb int64 = 12838682624
+	var ub int64 = 2872860672
 	assert.EqualValues(t, 1054023794, pgInfo["data_bytes"])
-	assert.EqualValues(t, 9965821952, pgInfo["avail_bytes"])
-	assert.EqualValues(t, 12838682624, pgInfo["total_bytes"])
-	assert.EqualValues(t, 2872860672, pgInfo["used_bytes"])
+	assert.EqualValues(t, avb, pgInfo["avail_bytes"])
+	assert.EqualValues(t, tb, pgInfo["total_bytes"])
+	assert.EqualValues(t, ub, pgInfo["used_bytes"])
```
Comment (jsoriano) [inline metricbeat/module/ceph/cluster_status/cluster_status_test.go:103]: Nit. No need to define additional variables:
```suggestion
	assert.EqualValues(t, int64(9965821952), pgInfo["avail_bytes"])
	assert.EqualValues(t, int64(12838682624), pgInfo["total_bytes"])
	assert.EqualValues(t, int64(2872860672), pgInfo["used_bytes"])
```

File: metricbeat/module/etcd/metrics/_meta/testdata/metrics.plain [hunk 2]
```diff
@@ -207,9 +207,9 @@ etcd_disk_backend_snapshot_duration_seconds_sum 0
 etcd_disk_backend_snapshot_duration_seconds_count 0
 # HELP etcd_disk_wal_fsync_duration_seconds The latency distributions of fsync called by wal.
 # TYPE etcd_disk_wal_fsync_duration_seconds histogram
-etcd_disk_wal_fsync_duration_seconds_bucket{le="0.001"} 0
-etcd_disk_wal_fsync_duration_seconds_bucket{le="0.002"} 0
-etcd_disk_wal_fsync_duration_seconds_bucket{le="0.004"} 0
+etcd_disk_wal_fsync_duration_seconds_bucket{le="0.001"} 1
+etcd_disk_wal_fsync_duration_seconds_bucket{le="0.002"} 1
+etcd_disk_wal_fsync_duration_seconds_bucket{le="0.004"} 1
```
Comment (jsoriano) [inline metricbeat/module/etcd/metrics/_meta/testdata/metrics.plain:212]: Umm, what error were causing the zero values? I think we should keep some of them in the test case.

File: metricbeat/module/etcd/metrics/metrics_test.go [hunk 3]
```diff
@@ -15,7 +15,7 @@
 // specific language governing permissions and limitations
 // under the License.
 
-// +build !integration
+// +build !integration, !386
```
Comment (jsoriano) [inline metricbeat/module/etcd/metrics/metrics_test.go:18]: Are these tests failing in all 32-bit platforms or only in Windows?

File: metricbeat/module/system/test_system.py [hunk 4]
```diff
@@ -586,6 +586,7 @@ def test_socket_summary(self):
 
             assert isinstance(udp["all"]["count"], int)
 
+    @unittest.skipIf(sys.platform == "win32", "Flaky test")
```
Comment (jsoriano) [inline metricbeat/module/system/test_system.py:589]: Flaky tests should have a open issue to address them in the future.

File: metricbeat/module/etcd/metrics/_meta/testdata/metrics.plain (hunk 2 above)
Comment (narph) [reply metricbeat/module/etcd/metrics/_meta/testdata/metrics.plain:212]: I mentioned above, when using the `expfmt` package to read and parse the values ending with 0 they are missing from the buckets. This happens only on windows 32 bit not all windows versions. I have updated the fields to 1 so we can have one of these tests still running on windows 32. There is a second test at the module level which will test with the `0` values as well (which has been skipped on windows 32 bit) but will run on all os's so we have that covered both ways.

File: metricbeat/module/etcd/metrics/metrics_test.go (hunk 3 above)
Comment (narph) [reply metricbeat/module/etcd/metrics/metrics_test.go:18]: only windows 32 bit

File: metricbeat/module/system/test_system.py (hunk 4 above)
Comment (narph) [reply metricbeat/module/system/test_system.py:589]: I would have created one separately but I had around 40+ tests, it would have been hard to manage the tickets, I have taken screenshots with all the ones that kept on popping up

File: metricbeat/module/ceph/cluster_status/cluster_status_test.go (hunk 1 above)
Comment (narph) [reply metricbeat/module/ceph/cluster_status/cluster_status_test.go]: I have updated the code

**Comment by v1v** [comment]
I've just added all the labels for all the `windows` that we support to validate there are no any regressions.

/test

File: metricbeat/module/etcd/metrics/_meta/testdata/metrics.plain (hunk 2 above)
Comment (jsoriano) [reply metricbeat/module/etcd/metrics/_meta/testdata/metrics.plain:212]: Interesting. But sounds good if we have this covered in other tests :+1:

File: metricbeat/module/etcd/metrics/metrics_test.go (hunk 3 above)
Comment (jsoriano) [reply metricbeat/module/etcd/metrics/metrics_test.go:18]: Could we explicitly skip the tests only for Windows 32 bits instead of for all 386 platforms? With a comment to know what they are skipped.

File: metricbeat/module/system/test_system.py (hunk 4 above)
Comment (jsoriano) [reply metricbeat/module/system/test_system.py:589]: Umm, does this test always fail in win32? If it always fail maybe instead of skipping it because of flakiness we can say in the comment that this test (or this feature) is not supported in win 32.

Comment (narph) [reply metricbeat/module/system/test_system.py:589]: the test does not always fail, it passed previously.

File: metricbeat/module/etcd/metrics/metrics_test.go (hunk 3 above)
Comment (narph) [reply metricbeat/module/etcd/metrics/metrics_test.go]: sure, I have made the changes

File: metricbeat/module/system/test_system.py (hunk 4 above)
Comment (jsoriano) [reply metricbeat/module/system/test_system.py:589]: If we plan to fix these flakiness at some moment we should have an issue for that, and reference it in the comment. Flaky tests can be hiding bugs.
But possibly it doesn't make much sense to dedicate a lot of effort to corner cases in 32-bits Windows :slightly_smiling_face:

**Review by jsoriano — APPROVED** [review]
Thanks for working on this!
